README.md

```diff
@@ -74,23 +79,54 @@ from within each directory.
+
+The Shark Walk worker also uses the [Streets of San Francisco Dataset](https://data.sfgov.org/Geographic-Locations-and-Boundaries/Streets-of-San-Francisco/4ugn-hf48) to create a record of each point in the polylines that describe the city streets. It saves these points, along with the intersections they are connected to, as a secondary graph that is used to find the 'entry' and 'exit' nodes during [pathfinding](#pathfinding).
+
+### Crime Risk
+
+<img src='readme/riskweight.png' width=800 alt="Crime risk weight per distance" />
+
+To build a crime risk model, the worker accesses the [SFPD Incidents Dataset](https://data.sfgov.org/Public-Safety/SFPD-Incidents-from-1-January-2003/tmnf-yvry) to retrieve criminal police reports from the last six months. For each node in the street graph, the worker computes a crime risk for that node by examining police reports in the vicinity. The closer a police report is to the node, the more it contributes to the risk at that node. A report that is at the node adds a value of one to the crime risk at the node. The weight of the police report on the crime risk falls off according to the modified sigmoid function shown above. At 100m from the node, a report adds 0.5 to the crime risk weight. At 250m from the node, a police report adds less that 0.001 to the crime risk weight. The worker ignores reports outside of 250m.
+
+## Pathfinding
+
+Armed with a graph of the city's streets and a risk level at each point, Shark Walk is able to develop deterministic optimal paths between two points in the city based on the weight given to the crime risk, or in other words, how far out of one's way one is willing to go to avoid crime risk.
+
+The first task is to compute the 'entry' and 'exit' nodes, which are the first and last nodes on the graph in the journey (this is necessary since the user's origin and destination will not exactly correspond to nodes in the Shark Walk geographic data model). Shark Walk selects the closest point on the street city street to the user's origin and destination from the secondary graph created by the worker. This point is connected to two intersection nodes and one is selected by Shark Walk to be either the entry node (close to the origin) or the exit node (close to the destination).
+
+Once the entry and exit nodes are selected, Shark Walk uses the [A* search algorithm](https://en.wikipedia.org/wiki/A*_search_algorithm) to find a pathway between these points. The cost function for moving between nodes is set as the distance traversed plus the risk traversed times a risk weight. The risk traversed is computed by multiplying the risk level at the next node times the distance to the next node. The A* heuristic is set to be the distance between the node and the destination plus an average risk level throughout the city times the distance times the risk weight.
```

Comment:
Is "risk weight" in the cost function of the A* algorithm here referring to how far someone is willing to walk out of their way to avoid risky areas? If so, I think we should explicitly state that to avoid confusion between this and the risk weight of each node. I usually think about the risk weight of each node when reading "risk weight."